### PR TITLE
Make Dimension.label source of truth for Dimension identity

### PR DESCRIPTION
**Pull request by philippjfr**
Implements https://github.com/holoviz/holoviews/issues/6260

### CODE REVIEW

File: holoviews/plotting/renderer.py [hunk 1]
```diff
@@ -433,6 +433,8 @@ def _render_ipywidget(self, plot):
             from IPython.display import display
             display(data, raw=True)
             return {'text/html': '<div style="display: none"></div>'}, {}
+        elif config.comms == 'colab':
+            load_notebook(config.inline)
```
Comment (hoxbro) [inline holoviews/plotting/renderer.py:437]: This seems to be unrelated to this PR :upside_down_face:

Comment (philippjfr) [reply holoviews/plotting/renderer.py:437]: Indeed :)

Comment (philippjfr) [inline holoviews/plotting/renderer.py:437]:
```suggestion
```

**Review by hoxbro — COMMENTED** [review]
My rough understanding is that `dim.name` will be used for data access, e.g., looking up a column in a DataFrame and `dim.label` will be used by HoloViews for ranges etc. Is this an ok (rough) understanding?

How thoroughly have you checked the codebase for this find and replace? The code still seems to use `name` for range lookup. 

https://github.com/holoviz/holoviews/blob/c8e7f5f887f05e74e6a8999c990dbd41265a086b/holoviews/plotting/util.py#L411-L411
https://github.com/holoviz/holoviews/blob/c8e7f5f887f05e74e6a8999c990dbd41265a086b/holoviews/operation/normalization.py#L109-L109

Have you checked if `hvplot` / `GeoViews` need updating too?

**Comment by philippjfr** [comment]
>Is this an ok (rough) understanding?

Yes.

>How thoroughly have you checked the codebase for this find and replace? The code still seems to use name for range lookup.

Clearly not quite thoroughly enough, will do another more exhaustive check now.

>Have you checked if hvplot / GeoViews need updating too?

Yes.

**Comment by philippjfr** [comment]
Did miss quite a few. Will scan more thoroughly one more time. Not that this only affects plotting code, so e.g. this isn't affected:

```
holoviews/holoviews/operation/normalization.py

Line 109 in c8e7f5f

 return {d.name: element.range(d.name, self.data_range) 
```

**Comment by philippjfr** [comment]
Okay, quite confident now, thoroughly reviewed the plotting code and replaced all remaining cases.

**Comment by philippjfr** [comment]
I think the only way we'll catch any more cases will be if we merge and include the change in the RC release.

**Review by hoxbro — APPROVED** [review]
